app/javascript/pages/admin/Reporting/index.jsx

```diff
@@ -0,0 +1,208 @@
+    }
+
+    if (Object.prototype.hasOwnProperty.call(changes, 'selectedItem')) {
+      const nextSelectedItem = changes.selectedItem
+      setDropDownTempSelectedItem(nextSelectedItem)
+    }
+  }
+
+  const handleDropdownOnSelect = item => {
+    setReport(item)
+  }
+
+  return (
+    <AdminSection>
+      <div style={{ width: "100%" }}>
+        <ToolbarHeader>
+          <Dropdown
+              isOpen={dropDownIsOpen}
+              onSelect={handleDropdownOnSelect}
+              onStateChange={handleDropdownStateChange}
+            >
+            <Field style={{ width: 200 }}>
+              <FormLabel>{t('volunteer_portal.admin.tab.reporting.select_label')}</FormLabel>
+              <Select>
+                {
+                  report === null ? t('volunteer_portal.admin.tab.reporting.dropdown.none') : reportKeyTitleMap[report]
+                }
+              </Select>
+            </Field>
+            <Menu >
+              <Item value="user">{t('volunteer_portal.admin.tab.reporting.dropdown.users')}</Item>
+            </Menu>
+          </Dropdown>
+          <FormField>
+            <FormLabel>{t('volunteer_portal.admin.tab.reporting.filter')}</FormLabel>
+            <div style={{ marginTop: 8 }}>
+              {/* TODO: Local office filter? */}
+              <OfficeFilter/>
+            </div>
+          </FormField>
+          <div style={{ display: "flex" }}>
+            <FormField>
+                <FormLabel>{t('volunteer_portal.admin.tab.reporting.from')}</FormLabel>
+                <Datepicker value={dateRange.start} onChange={changeStartDate} maxValue={today} locale={i18n.language}>
+                  <CombinedInput left bare />
+                </Datepicker>
+            </FormField>
+            <FormField>
+                <FormLabel>{t('volunteer_portal.admin.tab.reporting.to')}</FormLabel>
+                <Datepicker value={dateRange.end} onChange={changeEndDate} maxValue={today} locale={i18n.language}>
+                  <CombinedInput right bare />
+                </Datepicker>
+            </FormField>
+          </div>
+          <a href={octetStream && `data:application/octet-stream;filename=export.csv,${octetStream}`} download="export.csv">
+            <Button disabled={!octetStream}>{t('volunteer_portal.admin.tab.reporting.exportascsv')}</Button>
+          </a>
+        </ToolbarHeader>
+        {
+          report === 'user' ?
+          (
+            userLoading ?
+            <div>
+              <Skeleton height="46px" />
+              {
+                R.range(0, 10).map((_, index) => <Skeleton key={index} height="38px" />)
+              }
+            </div>
+            :
+            <UserReporting
+              users={R.propOr([], 'users', userData)}
+              startDate={dateRange.start}
+              endDate={dateRange.end}
+              onStartChange={changeStartDate}
+              onEndChange={changeEndDate}
+              setOctectStream={setOctectStream}
+            />
```

Comment:
typically the loading display is handled by the component itself. In this case `UserReporting`, which exposes a prop `loading` and it will switch internally to show the stencil or content. 

The benefits are to increase conposability and reduce repetition:
1. you could move `UserReporting` anywhere and the loading display moves with it
2. you don't need to duplicate stencil code if you use the component in multiple places.

Also you remove the need for the nasty nested ternary operator.